src/Pulsar.Client/Internal/ClientCnx.fs

```diff
@@ -831,7 +832,10 @@ and internal ClientCnx (config: PulsarClientConfiguration,
                         | Result.Error (UnknownCommandType unknownType), consumed ->
                             Log.Logger.LogError("{0} UnknownCommandType {1}, ignoring message", prefix, unknownType)
                             reader.AdvanceTo consumed
-            with Flatten ex ->
+            with
+            | ?: OperationCancelException ->
```

Comment:
Can you please match after flatten, not before

Reply:
Sounds good - done.